src/Nethereum.RPC/TransactionReceipts/TransactionReceiptPollingService.cs

```diff
@@ -32,44 +32,44 @@ public TransactionReceiptPollingService(ITransactionManager transactionManager,
         }
 
         public Task<TransactionReceipt> SendRequestAndWaitForReceiptAsync(TransactionInput transactionInput,
-            CancellationTokenSource tokenSource = null)
+            CancellationToken token = default(CancellationToken))
         {
-            return SendRequestAndWaitForReceiptAsync(() => _transactionManager.SendTransactionAsync(transactionInput), tokenSource);
+            return SendRequestAndWaitForReceiptAsync(() => _transactionManager.SendTransactionAsync(transactionInput), token);
         }
 
         public Task<List<TransactionReceipt>> SendRequestsAndWaitForReceiptAsync(IEnumerable<TransactionInput> transactionInputs,
-            CancellationTokenSource tokenSource = null)
+            CancellationToken token = default(CancellationToken))
         {
             var funcs = new List<Func<Task<string>>>();
             foreach (var transactionInput in transactionInputs)
             {
                 funcs.Add(() => _transactionManager.SendTransactionAsync(transactionInput));
             }
-            return SendRequestsAndWaitForReceiptAsync(funcs.ToArray(), tokenSource);
+            return SendRequestsAndWaitForReceiptAsync(funcs.ToArray(), token);
         }
 
         public async Task<TransactionReceipt> SendRequestAndWaitForReceiptAsync(Func<Task<string>> transactionFunction,
-            CancellationTokenSource tokenSource = null)
+            CancellationToken token = default(CancellationToken))
         {
             var transaction = await transactionFunction().ConfigureAwait(false);
-            return await PollForReceiptAsync(transaction, tokenSource).ConfigureAwait(false);
+            return await PollForReceiptAsync(transaction, token).ConfigureAwait(false);
         }
 
-        public async Task<TransactionReceipt> PollForReceiptAsync(string transaction, CancellationTokenSource tokenSource = null)
+        public async Task<TransactionReceipt> PollForReceiptAsync(string transaction, CancellationToken token = default(CancellationToken))
         {
             var getTransactionReceipt = new EthGetTransactionReceipt(_transactionManager.Client);
             var receipt = await getTransactionReceipt.SendRequestAsync(transaction).ConfigureAwait(false);
             while (receipt == null)
             {
                 await Task.Delay(_retryMiliseconds).ConfigureAwait(false);
-                tokenSource?.Token.ThrowIfCancellationRequested();
+                token.ThrowIfCancellationRequested();
```

Comment:
This is actually the single imporant change (IIRC). The rest is just fixing compliation.